### PR TITLE
docs: update setup and build instructions

### PR DESCRIPTION
**Pull request by antonbaliasnikov**
# Code Review Checklist

## Purpose

Fixes README.md after LLVM 17 and builder updates:
1. Correct required tools list for each OS platform we support
2. Remove unnecessary anymore `--target` from the builder install
3. Combine verbose build instructions into sections

Easier to review using [rich diff](https://github.com/matter-labs/era-compiler-llvm/pull/520/files?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5).

## Ticket Number

N/A

## Requirements
- [ ] Have the requirements been met?
- [ ] Have stakeholder(s) approved the change?

## Implementation
- [ ] Does this code change accomplish what it is supposed to do?
- [ ] Can this solution be simplified?
- [ ] Does this change add unwanted compile-time or run-time dependencies?
- [ ] Could an additional framework, API, library, or service improve the solution?
- [ ] Could we reuse part of LLVM instead of implementing the patch or a part of it?
- [ ] Is the code at the right abstraction level?
- [ ] Is the code modular enough?
- [ ] Can a better solution be found in terms of maintainability, readability, performance, or security?
- [ ] Does similar functionality already exist in the codebase? If yes, why isn’t it reused?
- [ ] Are there any best practices, design patterns or language-specific patterns that could substantially improve this code? 

## Logic Errors and Bugs
- [ ] Can you think of any use case in which the
code does not behave as intended?
- [ ] Can you think of any inputs or external events
that could break the code?

## Error Handling and Logging
- [ ] Is error handling done the correct way?
- [ ] Should any logging or debugging information
be added or removed?
- [ ] Are error messages user-friendly?
- [ ] Are there enough log events and are they
written in a way that allows for easy
debugging?

## Maintainability
- [ ] Is the code easy to read?
- [ ] Is the code not repeated (DRY Principle)?
- [ ] Is the code method/class not too long?

## Dependencies
- [ ] Were updates to documentation, configuration, or readme files made as required by this change?
- [ ] Are there any potential impacts on other parts of the system or backward compatibility?

## Security
- [ ] Does the code introduce any security vulnerabilities?

## Performance
- [ ] Do you think this code change decreases
system performance?
- [ ] Do you see any potential to improve the
performance of the code significantly?

## Testing and Testability
- [ ] Is the code testable?
- [ ] Have automated tests been added, or have related ones been updated to cover the change?
	- [ ] For changes to mutable state
- [ ] Do tests reasonably cover the code change (unit/integration/system tests)? 
	- [ ] Line Coverage
	- [ ] Region Coverage
	- [ ] Branch Coverage
- [ ] Are there some test cases, input or edge cases
that should be tested in addition?

## Readability
- [ ] Is the code easy to understand?
- [ ] Which parts were confusing to you and why?
- [ ] Can the readability of the code be improved by
smaller methods?
- [ ] Can the readability of the code be improved by
different function, method or variable names?
- [ ] Is the code located in the right
file/folder/package?
- [ ] Do you think certain methods should be
restructured to have a more intuitive control
flow?
- [ ] Is the data flow understandable?
- [ ] Are there redundant or outdated comments?
- [ ] Could some comments convey the message
better?
- [ ] Would more comments make the code more
understandable?
- [ ] Could some comments be removed by making the code itself more readable?
- [ ] Is there any commented-out code?

## Documentation
- [ ] Is there sufficient documentation?
- [ ] Is the ReadMe.md file up to date?

## Best Practices
- [ ] Follow Single Responsibility principle?
- [ ] Are different errors handled correctly?
- [ ] Are errors and warnings logged?
- [ ] Magic values avoided?
- [ ] No unnecessary comments?
- [ ] Minimal nesting used?

## Experts' Opinion
- [ ] Do you think a specific expert, like a security
expert or a usability expert, should look over
the code before it can be accepted?
- [ ] Will this code change impact different teams, and should they review the change as well?

### CODE REVIEW

**Review by hedgar2017 — CHANGES_REQUESTED** [review]
@antonbaliasnikov could you do a favor and globally rename `zkEVM` -> `EraVM` in this PR?

**Comment by antonbaliasnikov** [comment]
> @antonbaliasnikov could you do a favor and globally rename `zkEVM` -> `EraVM` in this PR?

@hedgar2017 , sure, will do.

**Comment by FlashSheridan** [comment]
> >  could you do a favor and globally rename `zkEVM` -> `EraVM` in this PR?

But please keep that as a separate commit, if not a separate PR.  (I’ve already started reviewing the substantive changes…)

File: README.md [hunk 1]
```diff
@@ -28,46 +28,94 @@ The zkEVM back-end is called `EraVM`, and the architecture is called `eravm`.
...
+      # Install the required packages using pacman
+      pacman -Syu which cmake ninja curl git pkg-config clang lld
+      ```
 
-   3.a. `cargo install compiler-llvm-builder` on MacOS, or Linux for personal use
-   3.b. `cargo install compiler-llvm-builder --target x86_64-unknown-linux-musl` on Linux for distribution
+   * MacOS:
 
-   The builder is not the zkEVM LLVM framework itself, but a tool that clones its repository and runs the sequence of build commands.
-   By default it is installed in `~/.cargo/bin/`, which is recommended to be added to your `$PATH`.
```
Comment (FlashSheridan) [inline README.md:49]: Never mind; the text has been moved to a more appropriate place.
[I’m sorry to see these lines removed; that baffled me in my early Rust days.]

Comment (antonbaliasnikov) [reply README.md:49]: They are not removed, they are [here](https://github.com/matter-labs/era-compiler-llvm/pull/520/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R81).

**Comment by antonbaliasnikov** [comment]
> > > could you do a favor and globally rename `zkEVM` -> `EraVM` in this PR?
> 
> But please keep that as a separate commit, if not a separate PR. (I’ve already started reviewing the substantive changes…)

@FlashSheridan , yeah, I'm not touching it until you finalize the review, no problem 👍

File: README.md [hunk 2]
```diff
@@ -28,46 +28,94 @@ The zkEVM back-end is called `EraVM`, and the architecture is called `eravm`.
...
+      * Install [HomeBrew](https://brew.sh) package manager.
+      * Install the following packages:
```
Comment (FlashSheridan) [inline README.md:48]: For the other two platforms, you have only the comment in the code block, not duplicated by a preceding instruction.  Either makes sense, but we should be consistent.

File: README.md [hunk 3]
```diff
@@ -28,46 +28,94 @@ The zkEVM back-end is called `EraVM`, and the architecture is called `eravm`.
...
+      url = "https://github.com/matter-labs/era-compiler-llvm"
+      branch = "main"
+      ```
+
+</details>
+
+<details>
+<summary>5. Build LLVM.</summary>
+
+   * Clone and build the zkEVM LLVM framework using the `zkevm-llvm` tool:
```
Comment (FlashSheridan) [inline README.md]: See previous comment on whether we want instruction+comment in code block, or just comment in code block.

File: README.md [hunk 4]
```diff
@@ -28,46 +28,94 @@ The zkEVM back-end is called `EraVM`, and the architecture is called `eravm`.
...
+
+      The build artifacts will end up in the `./target-llvm/target-final/` directory.
+      You may point your `LLVM_SYS_170_PREFIX` to that directory to use this build as a compiler dependency.
+      If built with the `--enable-tests` option, test tools will be in the `./target-llvm/build-final/` directory, along   with copies of the build artifacts. For all supported build options, run `zkevm-llvm build --help`.
+
+</details>
 
 ## Troubleshooting
 
-- If you get a “failed to authenticate when downloading repository… if the git CLI succeeds then net.git-fetch-with-cli may help here” error,
```
Comment (FlashSheridan) [inline README.md:68]: Has this problem gone away?

**Review by FlashSheridan — CHANGES_REQUESTED** [review]
Other than the above, LGTM.  Much better organized, thanks.

File: README.md (hunk 4 above)
Comment (hedgar2017) [reply README.md:68]: I believe it was only relevant for private repos which we don't have anymore.

Comment (antonbaliasnikov) [reply README.md:68]: Yeah, exactly. There is no auth happening right now.

File: README.md (hunk 2 above)
Comment (antonbaliasnikov) [reply README.md:48]: I've decided to use the preceding instruction to allow users to directly copy the script content to their shells.

File: README.md (hunk 3 above)
Comment (antonbaliasnikov) [reply README.md]: Done, thx!

**Comment by antonbaliasnikov** [comment]
@hedgar2017 , @FlashSheridan 
Nice guys, thx for the review. I believe we're ready to merge this. I'll proceed with updating our other repositories.

P.S. I'll change `zkevm-llvm` to `zksync-llvm` later when the tool is ready.

File: README.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# zkSync Era: The zkEVM LLVM Framework
+# zkSync Era: The EraVM LLVM Framework
```
Comment (hedgar2017) [inline README.md]: Every `zkEVM` that is not about the target itself should become `ZKsync` per our discussion on Slack!

**Review by hedgar2017 — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (antonbaliasnikov) [reply README.md]: It should be ok now. thx!

File: README.md [hunk 6]
```diff
@@ -7,71 +7,116 @@ or decentralization. As it's EVM-compatible (with Solidity/Vyper), 99% of Ethere
...
 The project has multiple components. The core of the project is
 the `llvm` directory. This contains all of the tools, libraries, and header
 files needed to process intermediate representations and convert them into
 object files. Tools include an assembler, disassembler, bitcode analyzer, and
 bitcode optimizer. These tools are not yet officially supported for third-party front-ends.
-It also contains zkEVM modifications of the standard [LLVM regression tests](https://llvm.org/docs/TestingGuide.html#regression-tests).
+It also contains ZKsync modifications of the standard [LLVM regression tests](https://llvm.org/docs/TestingGuide.html#regression-tests).
 
-The zkEVM back-end is called `EraVM`, and the architecture is called `eravm`.
+The ZKsync back-end is called `EraVM`, and the architecture is called `eravm`.
```
Comment (hedgar2017) [inline README.md]: But this is about the target, so it must be EraVM.
It makes this sentence a tautology and probably redundant.

**Review by hedgar2017 — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (antonbaliasnikov) [reply README.md]: I was also thinking about that and wanted to check with you. I've removed this line.

**Review by hedgar2017 — APPROVED** [review]

**Review by FlashSheridan — APPROVED** [review]
Retracting outdated request for changes.
